### PR TITLE
Load & use DB config in recordsets

### PR DESCRIPTION
**Pull request by twostars**
### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [x] Make sure you are making a pull request against the **canary branch** (left side). Also you should start *your branch* off *our canary*.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

### Description
This behaves similarly to official, with the only difference it uses a shared method in CServerDlg (etc.) to construct the connection string from members of CServerDlg (etc.), whereas officially it accesses this externally, e.g.:

```cpp
CString strConnect;
strConnect.Format(..., m_pMain->m_strGameDSN, ...);
```

💔Thank you!

### CODE REVIEW

**Review by stevewgr — COMMENTED** [review]
Awesome work @twostars 🚀 and also thanks for your efforts in explaining how it was done officialy and wrongly xD I like your approach more with the cerntralized function to get the connection string, as well as the singleton approach. That will be also helpful in the long run as more code is added later.
Just one nit-pick comment, let me know if you have the patience to apply, otherwise I'll just merge it.

File: src/server/AIServer/AIServerDlg.cpp [hunk 1]
```diff
@@ -128,6 +132,10 @@ CServerDlg::CServerDlg(CWnd * pParent /*=NULL*/)
     m_byTestMode = 0;
     //m_ppUserActive = NULL;
     //m_ppUserInActive = NULL;
+
+    memset(m_strGameDSN, 0, sizeof(m_strGameDSN));
+    memset(m_strGameUID, 0, sizeof(m_strGameUID));
+    memset(m_strGamePWD, 0, sizeof(m_strGamePWD));
```
[the remark this answers is not in the document]
Comment (twostars) [reply src/server/AIServer/AIServerDlg.cpp]: Resolved by https://github.com/ko4life-net/ko/pull/226/commits/65c633fd5bd770b73945fde5fa37f563b40567b6

**Review by stevewgr — APPROVED** [review]
image

**Comment by stevewgr** [comment]
Looks like some stuff were missed in this PR and I fixed it here for the record: https://github.com/ko4life-net/ko/pull/268

in this commit: https://github.com/ko4life-net/ko/commit/1fa93c7933a12f6003a8ba93cf3cc3b0484e1746

**Pull request merged**